### PR TITLE
Bug 1234621 - Add basic client-side filtering to perfherder alerts

### PR DESCRIPTION
**Pull request by wlach**
[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1219)

### CODE REVIEW

File: ui/js/controllers/perf/alerts.js [hunk 1]
```diff
@@ -110,6 +111,41 @@ perf.controller('AlertsCtrl', [
...
+        function updateAlertVisibility() {
+            _.forEach($scope.alertSummaries, function(alertSummary) {
+                _.forEach(alertSummary.alerts, function(alert) {
+                    // only show alert if it passes all filter criteria
+                    alert.visible =
+                        (alert.series_signature.framework_id === $scope.filterOptions.framework.id) &&
+                        (!$scope.filterOptions.hideImprovements || alert.is_regression) &&
+                        _.every($scope.filterOptions.filter.split(' '),
+                                function(matchText) {
+                                    return !matchText ||
```
Comment (jmaher) [inline ui/js/controllers/perf/alerts.js]: why do we have !matchText ?

File: ui/js/controllers/perf/alerts.js [hunk 2]
```diff
@@ -284,27 +321,39 @@ perf.controller('AlertsCtrl', [
...
-        });
-
+        $q.all([PhFramework.getFrameworkList().then(
+            function(frameworks) {
+                $scope.frameworks = frameworks.data;
+            }),
+                ThRepositoryModel.get_list().then(function(response) {
+                    $scope.projects = response.data;
+                    $scope.selectedProject = _.findWhere($scope.projects, {
+                        name: thDefaultRepo ? thDefaultRepo : thDefaultRepo
```
Comment (jmaher) [inline ui/js/controllers/perf/alerts.js]: huh: "thDefaultRepo ? thDefaultRepo : thDefaultRepo" ?

Comment (wlach) [inline ui/js/controllers/perf/alerts.js]: This is old code, but it's obviously silly. I'll fix it.

File: ui/js/controllers/perf/alerts.js (hunk 1 above)
Comment (wlach) [inline ui/js/controllers/perf/alerts.js]: If there's no matchText, we shouldn't filter on it. So we check for its absence.

**Pull request merged**